src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/NamingStyles/NamingStyle.cs

```diff
@@ -63,6 +63,32 @@ internal partial struct NamingStyle
                 newName, newPrefix, newSuffix, newWordSeparator, newCapitalizationScheme);
         }
 
+        public override bool Equals(object obj)
+        {
+            return obj is NamingStyle other
+                && Equals(other);
+        }
+
+        public bool Equals(NamingStyle other)
+        {
+            return ID == other.ID
+                && Name == other.Name
+                && Prefix == other.Prefix
+                && Suffix == other.Suffix
+                && WordSeparator == other.WordSeparator
+                && CapitalizationScheme == other.CapitalizationScheme;
+        }
+
+        public override int GetHashCode()
+        {
+            return Hash.Combine(ID.GetHashCode(),
+                Hash.Combine(Name?.GetHashCode() ?? 0,
+                    Hash.Combine(Prefix?.GetHashCode() ?? 0,
+                        Hash.Combine(Suffix?.GetHashCode() ?? 0,
+                            Hash.Combine(WordSeparator?.GetHashCode() ?? 0,
+                                (int)CapitalizationScheme)))));
```

Comment:
is HashCode.Combine not available at this layer?

Reply:
HashCode.Combine is not in any layer IIRC (netcoreapp only)